ops/model.py

```diff
@@ -507,7 +507,13 @@ class ModelBackend:
     def __init__(self):
         self.unit_name = os.environ['JUJU_UNIT_NAME']
         self.app_name = self.unit_name.split('/')[0]
+        self.function_name = os.environ.get('JUJU_FUNCTION_NAME',
```

Comment:
This seems awkward. It breaks the idea that the model and the backend may be used for multiple events. Let's please try to keep the awkwardness of function handling into the function logic itself somehow. A cheap way for now is moving the environment checking into the function event itself. This is just a sanity check either way as far as I understand it, so shouldn't be a problem.

Reply:
Yes, just a sanity check. I'll move it as you described.